components/encoder/ams/ams_as5048.go

```diff
@@ -151,28 +171,21 @@ func (enc *Encoder) Reconfigure(
 	if err != nil {
 		return err
 	}
-
-	brd, err := board.FromDependencies(deps, newConf.BoardName)
-	if err != nil {
-		return err
-	}
-	localBoard, ok := brd.(board.LocalBoard)
-	if !ok {
-		return errors.Errorf(
-			"board with name %s does not implement the LocalBoard interface", newConf.BoardName,
-		)
-	}
-	i2c, exists := localBoard.I2CByName(newConf.I2CBus)
-	if !exists {
-		return errors.Errorf("unable to find I2C bus: %s", newConf.I2CBus)
-	}
 	enc.mu.Lock()
 	defer enc.mu.Unlock()
-	if enc.i2cBus == i2c || enc.i2cAddr == byte(newConf.I2CAddr) {
-		return nil
+
+	if enc.i2cBusName != newConf.I2CBus || enc.i2cBus == nil {
+		bus, err := genericlinux.NewI2cBus(newConf.I2CBus)
+		if err != nil {
+			msg := fmt.Sprintf("can't find I2C bus '%q' for AMS encoder", newConf.I2CBus)
```

Comment:
@penguinland  Just a note that I noticed when testing- the `genericlinux.NewI2cBus` call does **not** error if the i2c bus does not exist, so this error message would never be hit. Is this expected? In this implementation, if I put an invalid i2cBus, it successfully reconfigures and the code errors when calling `openHandle` in `resetPositon`. In the old implementation an invalid i2c bus fails reconfiguration at the `localBoard.I2CByName` call. Is this an issue?

Reply:
Hm... good question. The distinction is that the old way relied on the board config naming the I2C bus, and would fail if the name didn't exist. but if you named a nonexistent bus, I think this part would not have errors and then you'd get errors later when you actually try using the bus. I think that's analogous to the current behavior. 

It could be convenient to have `NewI2cBus` run some kind of check and return errors if the bus doesn't exist, but off the top of my head I don't see what that check would look like: we don't know which address to talk to (perhaps multiple addresses?) nor what we'd say to them. We could try creating something reminiscent of `i2cdetect`, but that sounds like a giant pain.

So, I'm inclined to say that this isn't an issue, but I wish we could more reliably return an error here. :confused:

Reply:
let's check with the ina219 sensor which relies completely on an external library to open its i2c bus

Reply:
The ina219 uses [this function](https://github.com/d2r2/go-i2c/blob/master/i2c.go#L31-L41), which immediately tries opening the device (file in `/dev`) and telling it what address we're going to connect to. It returns an error if the device cannot be opened or written to. 

Part of me thinks we should change `genericlinux.NewI2cBus` to use the same library the ina219 uses: it would be more consistent to use just one I2C library throughout the RDK. Part of me thinks that updating that is nontrivial (specifically, the ina's library expects the bus ID to be an int, while genericlinux/Periph's version wants a string that can include a `"/dev/i2c-"` prefix), and the consistency is not worth a breaking change.

If we do want to change that up, I think it's a separate enough issue to warrant its own ticket/PR, and the current work should not be blocked by it.

Reply:
As a stopgap measure to split the difference, we could add into `genericlinux.NewI2cBus` a thing to get the longest numeric suffix of the string passed in, prepend a `/dev/i2c-` to the front, and try opening that file, just to see if we get permission errors. It won't work if you're passing in a non-numeric string (e.g., you've set up udev to create semantic symlinks like `/dev/i2c-humidity` and put that in the config), but I doubt many people do that (and we can check in the config database to see how common it really is). but again, I think changing that should be a separate PR

Reply:
Agree that changing that is a separate pr and separate ticket altogether, but erroring out at the i2c/spi handle opening step is important.